social-tw/social-tw-website
# [WIP] Feature/issue 148 implement vote feature

Please follow the guidelines in [Contribution.md](https://github.com/social-tw/social-tw-website/blob/main/CONTRIBUTING.md) first, then start to create your pull request here.

## Summary

Implemented the voting feature based on the Figma design. Also integrated with the backend API for voting actions.

## Linked Issue

fix #148 

## Details

The voting feature has been implemented with buttons for upvote and downvote. Upon clicking these buttons, the corresponding backend API endpoint is called to register the vote. The current implementation ensures that a user can either upvote, downvote, or reset their vote.

## Impacted Areas

- `Post.tsx`: Added logic for handling voting actions and updating the UI.
- `useVotes.ts`: Integrated with the backend API for voting.
- `VoteAction.ts`: Vote action enum.

## Tests

Integration tests and unit tests are still pending for the voting feature. 

## Possible Impacts

Existing features related to posts might be affected by this change, especially any features that interact with post reactions or user actions.

## Visual Materials

No UI effects have been introduced outside of the Figma design, so no screenshots or videos are attached. 

## Verification Steps

1. Navigate to a post.
2. Click the upvote button to upvote a post.
3. The vote count should increase.
4. Clicking the upvote button again should reset the vote.
5. Clicking the downvote button after upvoting should switch the vote to a downvote.
6. The process should be similar for downvoting.

## Todo

-   [ ] Integration tests for voting feature.
-   [ ] Unit tests for voting functionality.

## Checklist

-   [ ] All new and existing tests pass (pending new tests)
